pkg/selector/selector.go

```diff
@@ -528,6 +529,11 @@ func IsAllowedNamespaces(ctx context.Context, c client.Client, namespace string)
 	return false, nil
 }
 
+func getRandomNumber(length int) int64 {
```

Comment:
`max` or `upperbound` seem to be a better name than `length`. And since you never get a negative value from `rand`, what about using `uint64` instead of `int64`?

Reply:
sure makes sense